snuba/datasets/sessions_processor.py

```diff
@@ -30,6 +31,77 @@ def process_message(self, message, metadata) -> Optional[ProcessedMessage]:
         # some old relays accidentally emit rows without release
         if message["release"] is None:
             return None
+
+        received = _ensure_valid_date(datetime.utcfromtimestamp(message["received"]))
+        if received is None:
+            metrics.increment("empty_received_date")
+
+        if message.get("aggregates") is not None:
+            template = {
+                "session_id": NIL_UUID,
+                "seq": 0,
+                "org_id": message["org_id"],
+                "project_id": message["project_id"],
+                "retention_days": message["retention_days"],
+                "duration": MAX_UINT32,
+                "received": received if received is not None else datetime.now(),
+                "release": message["release"],
+                "environment": message.get("environment") or "",
+            }
+
+            batch = []
+            for group in message["aggregates"]:
+                started = _ensure_valid_date(
+                    datetime.utcfromtimestamp(group["started"])
+                )
+                if started is None:
+                    continue
+
+                group_template = dict(
+                    template,
+                    distinct_id=str(uuid.UUID(group.get("distinct_id") or NIL_UUID)),
+                    started=started,
+                )
+
+                if group.get("exited") is not None:
+                    batch.append(
+                        dict(
+                            group_template,
+                            status=STATUS_MAPPING["exited"],
+                            errors=0,
+                            quantity=group["exited"],
+                        )
+                    )
+                if group.get("errored") is not None:
+                    batch.append(
+                        dict(
+                            group_template,
+                            status=STATUS_MAPPING["errored"],
+                            errors=1,
+                            quantity=group["errored"],
+                        )
+                    )
+                if group.get("abnormal") is not None:
+                    batch.append(
+                        dict(
+                            group_template,
+                            status=STATUS_MAPPING["abnormal"],
+                            errors=1,
+                            quantity=group["abnormal"],
+                        )
+                    )
+                if group.get("crashed") is not None:
+                    batch.append(
+                        dict(
+                            group_template,
+                            status=STATUS_MAPPING["crashed"],
+                            errors=1,
+                            quantity=group["crashed"],
+                        )
+                    )
+
+            return InsertBatch(batch)
```

Comment:
talked with jan, we will move this exploding into relay rather, so each kafka message basically corresponds to a single row in the raw table.